src/main/java/com/amazon/opendistroforelasticsearch/security/privileges/PrivilegesEvaluator.java

```diff
@@ -338,19 +337,19 @@ public PrivilegesEvaluatorResponse evaluate(final User user, String action0, fin
 
         if(privilegesInterceptor.getClass() != PrivilegesInterceptor.class) {
 
-            final Boolean replaceResult = privilegesInterceptor.replaceKibanaIndex(request, action0, user, dcm, requestedResolved, mapTenants(user, mappedRoles));
+            final PrivilegesInterceptor.ReplaceResult replaceResult = privilegesInterceptor.replaceKibanaIndex(request, action0, user, dcm, requestedResolved, mapTenants(user, mappedRoles));
 
             if(log.isDebugEnabled()) {
                 log.debug("Result from privileges interceptor: {}", replaceResult);
             }
 
-            if (replaceResult == Boolean.TRUE) {
-                auditLog.logMissingPrivileges(action0, request, task);
-                return presponse;
-            }
-
-            if (replaceResult == Boolean.FALSE) {
-                presponse.allowed = true;
+            if (!replaceResult.continueEvaluation) {
+                if (replaceResult.accessDenied) {
+                    auditLog.logMissingPrivileges(action0, request, task);
+                } else {
+                    presponse.allowed = true;
+                    presponse.request = replaceResult.createIndexRequest;
+                }
```

Comment:
Is it possible to factor out to separate function for replacing kibana index to avoid duplication ?